dagmc_stats/dagmc_stats.py

```diff
@@ -297,3 +316,172 @@ def get_coarseness(my_core, meshset, entity_ranges, geom_dim):
+    vertex related data
+
+    outputs
+    -------
+    gc : gaussian curvature value of the vertex
+    """
+    vert_entries = tri_vert_data[tri_vert_data['vert'] == vert_i]
+    sum_alpha_angles = sum(vert_entries['angle'])
+    gc = np.abs(2 * np.pi - sum_alpha_angles)
+    return gc
+
+
+def get_lri(vert_i, gc_all, tri_vert_data, my_core):
+    """Get local roughness value of a vertex
+    Reference: https://www.sciencedirect.com/science/article/pii/
+    S0097849312001203
+    Formula 2, 3
+
+    inputs
+    ------
+    vert_i : vertex entity handle
+    gc_all : dictionary in the form of vertex : gaussian curvature
+    value of the vertex
+    tri_vert_data : numpy structured array that stores the triangle
+    and vertex related data
+    my_core : a MOAB Core instance
+
+    outputs
+    -------
+    Lri : local roughness value of the vertex
+    """
+    DIJgc_sum = 0
+    Dii_sum = 0
+    vert_j_list = list(my_core.get_adjacencies(my_core.get_adjacencies(
+        vert_i, 2, op_type=0), 0, op_type=1))
+    vert_j_list.remove(vert_i)
+    for vert_j in vert_j_list:
+        # get tri_ij_list (the list of the two triangles connected to both
+        # vert_i and vert_j)
+        tri_i_list = my_core.get_adjacencies(vert_i, 2, op_type=0)
+        tri_j_list = my_core.get_adjacencies(vert_j, 2, op_type=0)
+        tri_ij_list = list(set(tri_i_list) & set(tri_j_list))
+        # get vertices that are in triangles in tri_ij_list
+        # there will be six elements representing four vertices
+        # vert_i and vert_j will appear twice
+        four_vert_0 = tri_vert_data[tri_vert_data['tri'] == tri_ij_list[0]]
+        four_vert_1 = tri_vert_data[tri_vert_data['tri'] == tri_ij_list[1]]
+        four_vert = np.concatenate([four_vert_0, four_vert_1])
+
+        # get beta angles
+        # remove the verts that are shared by two triangles (that will appear
+        # twice)
+        beta_angles = []
+        for entry in four_vert:
+            matched = four_vert[four_vert['vert'] == entry['vert']]
+            if len(matched) < 2:
+                beta_angles.append(entry['angle'])
+
+        Dij = 0.5 * (1/np.tan(beta_angles[0]) + 1/np.tan(beta_angles[1]))
+        DIJgc_sum += (Dij * gc_all[vert_j])
+        Dii_sum += Dij
+    Lri = abs(gc_all[vert_i] - DIJgc_sum/Dii_sum)
+    return Lri
+
+
+def get_roughness(my_core, native_ranges):
+    """Get local roughness values of all the non-isolated vertices
+
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    native_ranges : a dictionary containing ranges for each native type
+    in the file (VERTEX, TRIANGLE, ENTITYSET)
+
+    outputs
+    -------
+    roughness : (list) the roughness for all surfaces in the meshset
+    """
+    tri_vert_data, all_verts = get_tri_vert_data(my_core, native_ranges)
+    gc_all = get_gaussian_curvature(my_core, native_ranges, all_verts,
+                                    tri_vert_data)
+    roughness = []
```

Comment:
Should this be a numpy array (should we always prefer numpy arrays)?  Perhaps review other places in the code where lists are used and decide if they should numpy arrays?

Reply:
Related question - should we pre-allocate this to be long enough for all data and then populate it rather than appending.  This can lead to memory resize events that can be slow when the size of `all_verts` gets large (e.g. O(10^6))

Reply:
```suggestion
    roughness = np.zeros(len(all_verts))
```